SocialCareCaseViewerApi.Tests/V1/UseCase/Residents/GetAllUseCaseTests.cs

```diff
@@ -0,0 +1,185 @@
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+using SocialCareCaseViewerApi.V1.Boundary.Response;
+using SocialCareCaseViewerApi.V1.Gateways;
+using SocialCareCaseViewerApi.V1.Infrastructure;
+using SocialCareCaseViewerApi.V1.UseCase;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SocialCareCaseViewerApi.Tests.V1.UseCase.Residents
+{
+    [TestFixture]
+    public class GetAllUseCaseTests
+    {
+        private GetAllUseCase _getAllUseCase;
+        private Mock<IDatabaseGateway> _mockDatabaseGateway;
+        private Fixture _fixture = new Fixture();
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockDatabaseGateway = new Mock<IDatabaseGateway>();
+            _getAllUseCase = new GetAllUseCase(_mockDatabaseGateway.Object);
+
+            _mockDatabaseGateway.Setup(x => x.GetResidentsBySearchCriteria(
+                 It.IsAny<int>(),
+                 It.IsAny<int>(),
+                 It.IsAny<long>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>())
+            ).Returns(new List<ResidentInformation>());
+        }
+
+        [Test]
+        public void ReturnsEmptyListWhenMosaicIdIsProvidedInLettersOnlyFormat()
+        {
+            var request = new ResidentQueryParam() { MosaicId = "nonNumeric" };
+
+            _getAllUseCase.Execute(request, cursor: 0, limit: 20)
+                .Should()
+                .BeEquivalentTo(new ResidentInformationList() { Residents = new List<ResidentInformation>() });
+        }
+
+        [Test]
+        public void IfLimitLessThanTheMinimumWillUseTheMinimumLimit()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetResidentsBySearchCriteria(0, 10, null, null, null, null, null, null, null))
+                .Returns(new List<ResidentInformation>()).Verifiable();
+
+            _getAllUseCase.Execute(new ResidentQueryParam(), 0, 4);
+
+            _mockDatabaseGateway.Verify();
+        }
+
+        [Test]
+        public void IfLimitMoreThanTheMaximumWillUseTheMaximumLimit()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetResidentsBySearchCriteria(0, 100, null, null, null, null, null, null, null))
+                .Returns(new List<ResidentInformation>()).Verifiable();
+
+            _getAllUseCase.Execute(new ResidentQueryParam(), 0, 400);
+
+            _mockDatabaseGateway.Verify();
+        }
+
+        [Test]
+        public void ReturnsTheNextCursor()
+        {
+            var residents = _fixture.CreateMany<ResidentInformation>(10);
+            int idCount = 10;
+            foreach (ResidentInformation resident in residents)
+            {
+                idCount++;
+                resident.MosaicId = idCount.ToString();
+            }
+            var expectedNextCursor = residents.Max(r => r.MosaicId);
```

Comment:
Could you explain what we're doing here in the setup? Why do we need to change the `MosaicId`s of the residents?

Reply:
When fixture library creates those person objects it uses Ids like MosaicId18e8cb3e-d6d8-4e4b-9165-3ad8413d38a5. That idCount together with the loop resets the IDs to values we use (long), so the next cursor calculation can be performed correctly and reflects the types we use.

Reply:
Ahhhhh okay, I see thanks!